data-platform/core-converged-db/oracle-database-23c-new-features/OracleJDBC23cNewFeatures/src/main/java/swierzy/oraclejdbc23newfeatures/Main.java

```diff
@@ -0,0 +1,156 @@
+package swierzy.oraclejdbc23newfeatures;
+
+import java.sql.*;
+import oracle.jdbc.*;
+
+import java.util.UUID;
+import java.util.concurrent.*;
+import oracle.rsi.*;
+import java.time.*;
+
+/*
+   main application class
+   it uses two additional subscriber classes
+   1. DMLSubscriber as a subscriber for the INSERT statement being called asynchronously
+   2. QuerySubscriber as a subscriber for the SELECT statement being called asynchronously
+   it provides two methods demonstrating JDBC 23c new features
+   1. reactiveCallsDemo() which demonstrates Oracle JDBC 23c driver support for reactive programming
+   2. streamsIngestionDemo() which demonstrates Oracje JDBC 23c driver support for streams ingestion
+ */
+public class Main {
+    private static CountDownLatch latch = new CountDownLatch(2);
+    /*
+       reactiveCallsDemo: method, which demonstrates reactive programming support
+       provided by Oracle23c JDBC driver
+     */
+    public static void reactiveCallsDemo() throws Exception {
+        // database connection used by psDML and psQuery PreparedStatement objects
+        Connection con;
+        // prepared statements used to call INSERT and SELECT SQL statements in NOWAIT mode
+        PreparedStatement psDML, psQuery;
+
+        // database connection creation
+        System.out.println("Main.reactiveCallsDemo: connecting to the database.");
+        con = DriverManager.getConnection(
+                "<db_url>",
```

Comment:
Don't repeat yourself and make it more flexible for the reader. Get these values from parameters.